File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        if(head.next ==null) return null;
        if(head.next.next == null){ 
            head.next = null;
            return head;
        }
        ListNode fast = head;
        ListNode slow = head;
        while(fast!=null && fast.next!=null){
            slow = slow.next;
            fast = fast.next.next;
        }
        // slow.val = slow.next.val;
        // slow.next = slow.next.next;
       ListNode del= slow.next;
        slow.val = del.val;
        slow.next = slow.next.next;
        del = null;
        return head;
    }
}

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
int findSize(ListNode* head) {
    int cnt = 0;
    while(head != NULL) {
        cnt++;
        head = head -> next;
    }
    return cnt;
}
public:
    ListNode* deleteMiddle(ListNode* head) {
        if(head -> next == NULL) return NULL;
        // if(head == NULL || head -> next == NULL) return NULL;
        int size = findSize(head);
        int middle = size / 2;
        // if(size % 2 == 0) {
        //     middle = (size / 2) +1;
        // } else {
        //     middle = size / 2;
        // }
        ListNode* curr = head;
        for(int i = 0; i < middle - 1; i++) {
            curr = curr -> next;
        }
        curr -> next = curr -> next -> next;
        return head;
    }
};